inst/www/htmlwidgets.js

```diff
@@ -660,7 +660,11 @@
   }
 
   // Wait until after the document has loaded to render the widgets.
-  if (document.addEventListener) {
+  if (shinyMode && window.jQuery.fn.jquery > "3.0.0") {
```

Comment:
How about:

```
if (shinyMode) {
```

or

```
if (window.jQuery) {
```

? I don't think we ever want to behave differently than jQuery, if jQuery is available.

Reply:
Wouldn't that change the order of execution if we are outside of Shiny but using jQuery (due to the `setTimeout`)? Maybe it's safer to do:

```
if (shinyMode && window.jQuery) {
```

Reply:
Now that I think of it, we could also do:

```
if (shinyMode) {
window.jQuery(document).on("shiny:bound", function() {
      window.HTMLWidgets.staticRender();
 });
}
```

Since that's what we'll likely suggest to shiny users?

Reply:
After talking with @wch and @schloerke we decided it's probably best to not use any of the existing shiny events and instead use `$(fn)`